spec/models/program_spec.rb

```diff
@@ -199,4 +199,21 @@
     end
   end
 
+  describe '#any_event_for_this_date?' do
+    it 'returns type should be boolean' do
+      expect(program.any_event_for_this_date?('').in?([true, false])).to eq(true)
+    end
+
+    it 'returns todays events if date is nil' do
+      date = DateTime.parse(' 00:00').utc
+      events = EventSchedule.where(schedule: program.selected_schedule).where(start_time: date..(date + 1))
+      expect(program.any_event_for_this_date?('')).to eq(events.any?)
```

Comment:
You need to test if the function is returning true, when there are events, and if it returns false, when there aren't any events. Unless you make sure that there are events, or there are not any events, you can't test this.